File: pet-clinic-scala/src/main/scala/com/github/rskupnik/repositories/PetRepository.scala
package com.github.rskupnik.repositories

import com.github.rskupnik.model.Pet
import org.springframework.data.repository.CrudRepository
import org.springframework.stereotype.Repository

// Need to specify java.lang.Long instead of Scala's Long

@Repository
trait PetRepository extends CrudRepository[Pet, java.lang.Long]

File: pet-clinic-java/src/main/java/com/github/rskupnik/petclinic/repositories/PetRepository.java
package com.github.rskupnik.petclinic.repositories;

import com.github.rskupnik.petclinic.model.Pet;
import org.springframework.data.repository.CrudRepository;
import org.springframework.stereotype.Repository;

@Repository
public interface PetRepository extends CrudRepository<Pet, Long> {
}
